Sources/PackageModel/Manifest.swift

```diff
@@ -165,21 +165,23 @@ public final class Manifest: ObjectIdentifierProtocol {
             return dependencies
         }
         #else
-        guard toolsVersion >= .v5_2 && productFilter != .everything else {
+        guard toolsVersion >= .v5_2 && packageKind != .root else {
             return dependencies
         }
-
-        var requiredDependencies: Set<PackageDependencyDescription> = []
-
+        
+        var requiredDependencyURLs: Set<String> = []
+        
         for target in targetsRequired(for: products) {
             for targetDependency in target.dependencies {
                 if let dependency = packageDependency(referencedBy: targetDependency) {
-                    requiredDependencies.insert(dependency)
+                    requiredDependencyURLs.insert(dependency.url)
                 }
             }
         }
-
-        return Array(requiredDependencies)
+        
+        let dependenciesByURL = Dictionary(dependencies.map({ ($0.url, $0) }), uniquingKeysWith: { $1 })
+        let requiredDependencies = requiredDependencyURLs.compactMap({ dependenciesByURL[$0] })
+        return requiredDependencies
```

Comment:
It looks as though the dependencies will still be returned in arbitrary order.  Should they be sorted by URL as they're converted from Set to Array?

Reply:
You're right, I wonder how this used to work? I actually went back to the old implementation [here](https://github.com/apple/swift-package-manager/blob/4022ff4c563eb0cddc54ced11293903a3b49100f/Sources/PackageModel/Manifest.swift#L272) (just swapping name for URL).

Reply:
In any case, I think we can just return `self.dependencies` (which is already sorted as desired) filtered by `requiredDependencyURLs`.